dash/_callback_context.py

```diff
@@ -247,5 +248,20 @@ def using_outputs_grouping(self):
     def timing_information(self):
         return getattr(flask.g, "timing_information", {})
 
+    @has_context
+    def set_props(self, component_id: typing.Union[str, dict], props: dict):
+        ctx_value = _get_context_value()
+        if isinstance(component_id, dict):
+            ctx_value.updated_props[json.dumps(component_id)] = props
+        else:
+            ctx_value.updated_props[component_id] = props
+
 
 callback_context = CallbackContext()
+
+
+def set_props(component_id: typing.Union[str, dict], props: dict):
```

Comment:
Is there a reason why the function signature for `set_props` isn't structured as 3 inputs: `component_id`, `prop_name`, `value`?

That to me seems more consistent with the rest of the Dash API.

Reply:
It is the same signature as the frontend `set_props`, we considered `set_props("id", prop=value)` but decided to stay with same API for frontend and backend.

Reply:
It's true, this pattern is a little more javascripty, but consistency is good and it could be convenient to be able to set multiple props of the same component in one call.